### ISSUE TITLE
reaction_network misses a basin when the PES has a zero/flat eigenmode (dedup counts zero-mode-displaced copies as distinct minima)

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/pounce#183]
## Summary

`reaction_network` fails to discover a basin when the PES has a zero (flat) eigenmode. The minima dedup compares full-coordinate Euclidean distance, so points that are the *same* minimum displaced along a flat direction are counted as distinct. They fill the `n_states` budget before the flooding search reaches other basins, so a whole basin (and the connections into it) is silently missed.

Real molecular systems always carry zero modes (rigid translation and rotation), so this affects any endpoint-free `reaction_network` run on a molecule, not just the toy below.

## Minimal reproduction (pure pounce + jax, no other deps)

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
import pounce

# Double well in x, harmonic in y, FLAT in z  -> z is a genuine zero eigenmode.
def V(v):
    x, y, z = v
    return (x**2 - 1.0)**2 + 2.0*y**2          # no z term

e    = jax.jit(V)
grad = jax.jit(jax.grad(V))
hess = jax.jit(jax.hessian(V))
e_np = lambda v: float(e(jnp.asarray(v)))
g_np = lambda v: np.asarray(grad(jnp.asarray(v)), float)
h_np = lambda v: np.asarray(hess(jnp.asarray(v)), float)

x0 = np.array([0.1, 0.2, 0.0])
net = pounce.reaction_network(e_np, x0, grad=g_np, hess=h_np,
                              n_states=4, n_transition_states=4, seed=0)

print("minima x   :", [round(float(m.x[0]), 3) for m in net.minima])
print("minima full:", [np.round(np.asarray(m.x), 4).tolist() for m in net.minima])
print("connections:", [tuple(int(k) for k in c.minima) for c in net.connections])
```

Output (identical for `seed` in 0, 1, 2, 3, 7):

```
minima x   : [1.0, 1.0, 1.0, 1.0]
minima full: [[1.0, 0.0, -0.0151], [1.0, 0.0, -0.0337], [1.0, 0.0, 0.0], [1.0, -0.0, 0.016]]
connections: [(2, -1), (-1, -1), (-1, -1), (-1, -1)]
```

All four "minima" are the same `x=+1` well, differing only in `z` by 0.015–0.034. The `x=-1` well is never found, so the `x=0` saddle's connection to it carries an unmapped `-1` endpoint.

## Root cause (confirmed)

The four minima differ by more than the default `dedup=0.01`, but only along the flat `z` direction, so dedup keeps them as distinct and they exhaust `n_states` before flooding reaches `x=-1`. Three checks isolate this:

1. **`dedup=0.3` recovers both wells** — `reaction_network(..., dedup=0.3)` returns `distinct_min_x=[-1.0, 1.0]`. The `-1` basin is reachable; only the dedup/budget interaction hides it.
2. **Removing the zero mode fixes it at the default dedup** — with `V += 2.0*z**2` (so `z` is no longer flat), the default `dedup=0.01` gives exactly two minima at `x=±1` and a fully-mapped connection `[(1, 0)]`.
3. **`find_critical_points` finds both wells** with the same seed/energy, because it does not fill a fixed `n_states` budget the same way.

## Suggested fix

Make minima dedup **mode-aware** so a flat direction cannot manufacture arbitrarily many "distinct" minima:

- Compare candidate minima in the non-zero-eigenvalue subspace, or quotient out known/near-null Hessian directions (rigid translations, rotations, or any eigenvalue below `eig_tol`) before the `dedup` distance test.
- Additionally, when a saddle branch descends (`descent_reach`) to a basin not yet in the minima set, register that basin rather than recording a `-1` endpoint, so connectivity is not lost when a basin is discovered via a saddle rather than via flooding.

## Environment

- pounce (pounce-solver) 0.7.0
- jax 0.10.2, numpy 2.5.1, Python 3.12, macOS arm64

Found while building an endpoint-free reaction-network workflow on top of `pounce.reaction_network`.
